Lesson 5-7 part 1-3:Chardev +procfs+ sysfs

MrStamp opened the pull request:
Implementation of simple module

MrStamp committed:
Create simple module part 1:Create empty files

Created two empty files Makefile and chrdev.c

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

MrStamp committed:
Create simple module part 2: Implement Makefile

Added instruction to Makefile to build module

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

MrStamp committed:
Create simple module part 3:Add includes to chrdev.c

Added includes to use command-line arguments.
Added major minor numbers and buffer size values

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

MrStamp committed:
Create simple module part 4 :Add file open option

Added function to check is file open.

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

MrStamp committed:
Create simple module part 5:Add release option

Added release function to close dev

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

MrStamp committed:
Create simple module part 6:Add read writte options and init exit options

Implements init exit functions and read writte options for module

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

galkinletter commented:
chrdev.c - static int count = 1;  Never used.

Please comment in  driver source.

galkinletter reviewed (changes requested):
chrdev.c - static int count = 1; Never used.
Please comment in driver source.

MrStamp committed:
Create simple module part 7:Add sysfs functional

Added sysfs cleanup

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

MrStamp committed:
Create simple module part 8:Add procfs functional

Added procfs status check
Fix unused parameter
Add empty liny in Makefile
Try to fix error handling

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

oleg-h2o commented on 02-KernelModules_CharacterDevice/chrdev.c:
```diff
@@ -0,0 +1,205 @@
+        return -ENOMEM;
+    	}
+    	error = sysfs_create_file(example_kobject, &sysfs_attribute.attr);
+    	if (error) {
```
Error handling is broken here.

oleg-h2o commented on 02-KernelModules_CharacterDevice/chrdev.c:
```diff
@@ -0,0 +1,205 @@
+//static int count = 1; 
+module_param(major, int, S_IRUGO);
+module_param(minor, int, S_IRUGO);
+module_param(count, int, S_IRUGO);
```
Unused parameter.

oleg-h2o reviewed (changes requested):
Plz fix. 
There are indentation problems in the file also.

galkinletter commented on 02-KernelModules_CharacterDevice/Makefile:
```diff
@@ -0,0 +1,9 @@
+	make -C $(BUILD_KERNEL) M=$(PWD) modules
+
+clean:
+	make -C $(BUILD_KERNEL) M=$(PWD) clean
```
Please add empty line at the end of file
```suggestion
	make -C $(BUILD_KERNEL) M=$(PWD) clean
	
```

galkinletter reviewed (changes requested):
Please fix:

- Error handling
- Code style
- Unused parameter

MrStamp committed:
Create simple module part 9 :Add bash script

Added bash script to test module
Fixed code style

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

galkinletter commented on 02-KernelModules_CharacterDevice/chrdev.c:
```diff
@@ -0,0 +1,205 @@
+}
+
+static int dev_proc_open(struct inode *inode, struct  file *file) {
+  return single_open(file, dev_proc_show, NULL);
```
Please Check kernel code with checkpatch and cppcheck

galkinletter reviewed (changes requested):
Please Check kernel code with checkpatch and cppcheck

MrStamp committed:
Create simple module part 10 : Provide test scenario

Signed-off-by: Mark Rubinshteyn <mark.rubinshteyn@meta.ua>

galkinletter commented:
It can be seen that the work was done
